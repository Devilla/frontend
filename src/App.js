import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
