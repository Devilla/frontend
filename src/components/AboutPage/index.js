import React, { Component } from 'react';
import {Animated} from "react-animated-css";
export default class About extends Component {
     componentDidMount(){
     window.scrollTo(0, 0)
   } 
  render() {
    return (
        <div>
            <div className="page-header"><div className="page-header-overlay">
            <div className="centered page-header-container w-container">
            <Animated className="page-header-title" animationIn="fadeInDown" animationOut="fadeOut" isVisible={true}>About us</Animated>
            </div></div></div>
            <div className="section innerpage">
                <div className="container w-container">
                    <p>We started Influence with one mission: to cut down the advertising cost and to convert more customers on your website.</p>
            <p>We use your existing customers’ data on your website and show it to your visitors so that they are socially influenced by buying behavior of people on the website.</p>
            <p>Using this tool we’ve been able to help a lot of businesses in increasing their money making and get more conversions.</p>
            <p>Social influence is the next big thing in marketing for influencing buyers to buy more.</p>
            <p>Let’s start with your free to start trial and get making more money for you.
            </p>
                   
                </div>
              </div>
        </div>
    );
  }
}
