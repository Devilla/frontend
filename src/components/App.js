import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
