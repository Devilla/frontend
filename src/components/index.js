export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
