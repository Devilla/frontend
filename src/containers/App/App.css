@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
