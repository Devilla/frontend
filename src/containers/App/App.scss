@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
