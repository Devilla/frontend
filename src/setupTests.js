import 'jest-enzyme';

window.matchMedia = window.matchMedia || function() {
  return {
    matches : false,
    addListener : function() {},
    removeListener: function() {}
  };
};
