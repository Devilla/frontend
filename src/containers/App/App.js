import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
