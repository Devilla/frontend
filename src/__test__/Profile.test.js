import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import { Profile } from 'components';
import { store } from 'App';
import '../setupTests';


// Snapshot for Terms React Component
describe('>>>T E R M S --- Snapshot',()=>{
    it('+++capturing Snapshot of terms', () => {
        const renderedValue =  renderer.create(<Profile store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});

describe("+++ test for the website pricing component", () => {
    let wrapper;
    beforeEach ( () => {
         wrapper = shallow(<Profile store={store}/>);
    })


    it('+++ render the DUMB component', () => {
      expect(wrapper.length).toEqual(1)
      })

      it('+++ contains website - container', () => {
        let wrapper = shallow(<Profile  store={store}/>);
        expect(wrapper.find('.container').length).toBe(0);
      });

      it('+++ contains website - input', () => {
        let wrapper = shallow(<Profile  store={store}/>);
        expect(wrapper.find('input').length).toBe(0);
      });

      it('+++ contains website - image', () => {
        let wrapper = shallow(<Profile  store={store}/>);
        expect(wrapper.find('img').length).toBe(0);
      });

      it('+++ contains website - button', () => {
        let wrapper = shallow(<Profile  store={store}/>);
        expect(wrapper.find('button').length).toBe(0);
      });
})
