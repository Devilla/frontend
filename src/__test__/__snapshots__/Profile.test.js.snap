// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>T E R M S --- Snapshot +++capturing Snapshot of terms 1`] = `
<div
  className="content fill profile-container"
>
  <div
    className="container-fluid"
  >
    <div
      className="col-sm-12"
    >
      <div
        className="profile-user-box card-box"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-2"
          >
            <span
              className="pull-left mr-3"
            >
              <img
                alt="User profile"
                className="thumb-lg rounded-circle"
                src="https://x1.xingassets.com/assets/frontend_minified/img/users/nobody_m.original.jpg"
              />
            </span>
            <div
              className=" media-body text-white"
            >
              <h4
                className="mt-1 mb-1 font-18"
              >
                Username
              </h4>
              <p
                className="text-light mb-0"
              >
                Country
              </p>
            </div>
          </div>
          <div
            className="col-sm-8"
          >
            <div
              className="card-box tilebox-one text-center  "
            >
              <h6
                className="text-muted text-uppercase mt-0"
              >
                Plan Type 
              </h6>
              <h3>
                 
              </h3>
            </div>
          </div>
          <div
            className="col-md-2"
          >
            <div
              className="profbtn"
            >
              <button
                className="btn btn-block btn-primary waves-light waves-effect upgrade1"
                onClick={[Function]}
                type="button"
              >
                Upgrade
              </button>
              <div>
                 
                <br />
              </div>
              <button
                className="btn btn-block btn-primary waves-light waves-effect billing1"
                onClick={[Function]}
                type="button"
              >
                Billing
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <h4
          className="header-title m-b-20"
        >
          Personal Information
        </h4>
        <div
          className="panel-body"
        >
          <hr />
          <div
            className="text-left"
          >
            <form>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p mt-5"
                  >
                    <strong>
                      First Name :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <input
                      autoComplete="given-name"
                      className="form-control"
                      id="firstName"
                      onChange={[Function]}
                      placeholder="First Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      Last Name :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <input
                      autoComplete="family-name"
                      className="form-control"
                      id="lastName"
                      onChange={[Function]}
                      placeholder="Last Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      Phone :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <input
                      autoComplete="tel-national"
                      className="form-control"
                      id="phoneNumber"
                      onChange={[Function]}
                      placeholder="Phone Number"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      Email :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <input
                      autoComplete="email"
                      className="form-control"
                      disabled={true}
                      id="email"
                      placeholder="Email Address"
                      type="text"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      Address :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <input
                      autoComplete="address-line2"
                      className="form-control"
                      id="address"
                      onChange={[Function]}
                      placeholder="Billing Address"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      Country :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <select
                      autoComplete="country-name"
                      className="form-control"
                      id="formControlsSelect"
                      onChange={[Function]}
                      placeholder="Country Name"
                      value=""
                    >
                      <option
                        value={null}
                      >
                        Select Country
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      States :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <select
                      autoComplete="address-level1"
                      className="form-control"
                      id="formfBillinControlsSelect"
                      onChange={[Function]}
                      placeholder="States"
                      value=""
                    >
                      <option
                        value={null}
                      >
                        Select State
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      City :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <select
                      autoComplete="address-level2"
                      className="form-control"
                      id="formControlsSelect"
                      onChange={[Function]}
                      placeholder="City"
                      value=""
                    >
                      <option
                        value={null}
                      >
                        Select City
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-6"
                >
                  <span
                    className="text-muted font-13 p"
                  >
                    <strong>
                      Company :
                    </strong>
                     
                  </span>
                  <div
                    className="form-group"
                  >
                    <input
                      autoComplete="organization"
                      className="form-control"
                      id="companyName"
                      onChange={[Function]}
                      placeholder="Company Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="col-md-12"
              >
                <div
                  className="text-right save"
                >
                  <button
                    className="btn btn-primary waves-effect"
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="mdi mdi-account-settings-variant mr-1"
                    />
                      
                    Save Profile
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
